packages/web/components/Quest/QuestDetailsHeader.tsx

```diff
@@ -21,12 +43,74 @@ export const QuestDetailsHeader: React.FC<Props> = ({ quest }) => {
   const {
     repetition,
     cooldown,
-    // createdAt,
-    // externalLink: link,
     title,
     status,
+    quest_skills: questSkills,
+    quest_roles: questRoles,
   } = quest;
 
+  // Archive quest modal
+  const cancelRef = useRef<HTMLButtonElement>(null);
+  const archiveButtonRef = useRef<HTMLButtonElement>(null); // focus returns here when modal is closed
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const toast = useToast();
+  const router = useRouter();
+  const [, updateQuest] = useUpdateQuestMutation(); // Don't need to use updateQuestResult that's returned from useUpdateQuestMutation e.g const [updateQuestResult, updateQuest] = ...
+
+  // Original skills and roles are needed (required field) for the updateQuest mutation
+  // because the updateQuest mutation deletes existing skills/roles and then adds in those
+  // that were submitted via the form (either the existing skills/roles or changes to them)
+  // If we keep the original skills/roles in, the quest can still be unarchived if needed
+
+  // Roles objects
+  const rolesObjects = questRoles.map((r) => ({
```

Comment:
If these are only used in `setQuestAsArchived` then they should probably be defined there too. Keep in mind that anything defined in a component outside of a hook gets reinstantiated on every render

Reply:
Got it - with this (and the note about `useCallback`/`async`) the idea is that by doing it this way, the values won't be recalculated every re-render?